app/upgrades.go

```diff
@@ -533,3 +558,48 @@ func updateIbcMarkerDenomMetadata(ctx sdk.Context, app *App) {
 	})
 	ctx.Logger().Info("Done updating ibc marker denom metadata")
 }
+
+// upgradeIBC handles all IBC upgrade logic for each upgrade.
+func upgradeIBC(ctx sdk.Context, app *App) error {
+	if err := pruneIBCExpiredConsensusStates(ctx, app); err != nil {
+		return err
+	}
+	return upgradeToIBCv8(ctx, app)
+}
+
+// pruneIBCExpiredConsensusStates upgrades IBC from v6 to v8.
+func pruneIBCExpiredConsensusStates(ctx sdk.Context, app *App) error {
+	ctx.Logger().Info("Pruning expired consensus states for IBC")
+	_, err := ibctmmigrations.PruneExpiredConsensusStates(ctx, app.appCodec, app.IBCKeeper.ClientKeeper)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("unable to prune expired consensus states, error: %s", err))
+		return err
+	}
+	ctx.Logger().Info("Done pruning expired consensus states for IBC")
+	return nil
+}
+
+// upgradeToIBCv8 upgrades IBC from v6 to v8.
+// TODO: Remove with the umber handlers.
+func upgradeToIBCv8(ctx sdk.Context, app *App) error {
+	ctx.Logger().Info("Upgrading to IBCv8")
+	params := app.IBCKeeper.ClientKeeper.GetParams(ctx)
+	params.AllowedClients = append(params.AllowedClients, exported.Localhost)
+	app.IBCKeeper.ClientKeeper.SetParams(ctx, params)
+	ctx.Logger().Info("Done upgrading to IBCv8")
+	return nil
+}
+
+// Migrate to new ConsensusParamsKeeper
+// TODO: Remove with the umber handlers.
+func migrateParams(ctx sdk.Context, app *App) error {
```

Comment:
Can we call this `migrateBaseappParams`?

There's a lot of params that we'll need to migrate on top of these ones. I'm not yet sure if we'll want to do those using module migrations or directly in the upgrade. I do know the upgrade will need to make a call to update the bank params, though.

Reply:
I can rename this to migrateIBCParams since this is IBC specific.